### PR TITLE
Version 1 of SearchPage.js

### PR DESCRIPTION
**Pull request by ctyuucsdedu**
I have made changes to 4 different files. Please check if all of them make sense to the front end. 
1. Include SearchPage.js in index.html
2. Updated main.js with SearchPage() function so calling this function would refresh the page to search page.
3. Updated the search button of the navBar and added a listener to call the SearchPage() function
4. Created the SearchPage.js where it prints the html and css of the search page.

Don't merge until all reviewers are sure this is correct. Thank Tucker and Steven for working together!!

### CODE REVIEW

**Comment by ctyuucsdedu** [comment]
You can check it by clicking the search button.

https://spudly-f0411--pr194-kevin-yu-front-end-nvh0y9t5.web.app

**Review by TuckerFrandsen — APPROVED** [review]
Good job, has all the essential components. Now we can add in the search functionality.

**Review by mlgi — APPROVED** [review]
Great work on the component, it looks exactly like the figmas!
The page function and usage of the Router is according to spec, so good work on that. 
Everything looks good to me, we'll go ahead and make issues regarding the back-end.

File: source/js/main.js [hunk 1]
```diff
@@ -61,6 +62,20 @@ function recipesPage() {
   renderRecipes(Object.keys(storage.recipeData), recipeSection);
 }
 
+/**
+ * This function would replace the main with the search page including the filter and the body
+ */
+function SearchPage() {
```
Comment (mlgi) [inline source/js/main.js:68]:
```suggestion
function SearchPage(results) {
```

File: source/js/main.js [hunk 2]
```diff
@@ -61,6 +62,20 @@ function recipesPage() {
...
+  const main = document.querySelector("main");
+  // delete everyting in main
+  main.innerHTML = "";
+  // make a section displaying recipes
+  const searchPage = document.createElement("search-page");
+  searchPage.data = "";
```
Comment (mlgi) [inline source/js/main.js:74]:
```suggestion
  searchPage.data = results;
```

**Review by mlgi — CHANGES_REQUESTED** [review]
Woops, sorry for the bait. We just need minor changes to align to spec mentioned in #179

File: source/js/main.js (hunk 1 above)
Comment (ctyuucsdedu) [reply source/js/main.js:68]: I have added them. Do I start a new pull request?

Comment (mlgi) [reply source/js/main.js]: no need, if you commit them to this branch, this PR will automatically update

**Review by mlgi — APPROVED** [review]
Thanks for the swift change! Everything's good to me!

**Comment by ctyuucsdedu** [comment]
I think I should add a listener to the form such that when submitting the recipe name it also do the same thing as clicking the search button

**Comment by ctyuucsdedu** [comment]
I have added the listener in NavBar.js and it is working on liver server but not on github
https://spudly-f0411--pr194-kevin-yu-front-end-nvh0y9t5.web.app

**Comment by ctyuucsdedu** [comment]
It won't affect the previous feature. I think the new listener would be fine.

**Review by mlgi — APPROVED** [review]
> I have added the listener in NavBar.js and it is working on liver server but not on github

Nah don't worry it updated fine, you just have to wait for the github action to finish re-deploying and maybe wipe your cache for the page. 

It works fine on my end and is up to spec for now. We'll leave the remaining functionality for the back-end to implement. Thank you, kevin!

**Review by Steven-Christensen — APPROVED** [review]
great work Kevin

**Pull request merged**